### ISSUE TITLE
Travis Logging is Fucked

### ISSUE THREAD

**Issue by zachriggle** [Gallopsled/pwntools#987]
It looks like something weird happened to logging in Travis: https://travis-ci.org/Gallopsled/pwntools/builds/234590377

It looks like *all* logging is being sent to either stdout or stderr, and that it's all being printed by Travis.

```
Document: tubes/ssh
-------------------
Connecting to 127.0.0.1 on port 22
Connected (version 2.0, client OpenSSH_5.9p1)
Authentication (publickey) successful!
Connecting to 127.0.0.1 on port 22: Done
Starting remote process 'false' on 127.0.0.1
Opening new channel: 'for py in python2.7 python2 python; do test -x "$(which $py 2>&1)" && exec $py -c \'#!/usr/bin/env python2\nimport os, sys, ctypes, resource, platform, stat\nfrom collections import OrderedDict\nexe   = \'\\\'\'false\'\\\'\'\nargv  = [\'\\\'\'false\'\\\'\']\nenv   = None\n\nos.chdir(\'\\\'\'.\'\\\'\')\n\nif env is not None:\n    os.environ.clear()\n    os.environ.update(env)\n\ndef is_exe(path):\n    return os.path.isfile(path) and os.access(path, os.X_OK)\n\nPATH = os.environ.get(\'\\\'\'PATH\'\\\'\',\'\\\'\\\'\').split(os.pathsep)\n\nif os.path.sep not in exe and not is_exe(exe):\n    for path in PATH:\n        test_path = os.path.join(path, exe)\n        if is_exe(test_path):\n            exe = test_path\n            break\n\nif not is_exe(exe):\n    sys.stderr.write(\'\\\'\'3\\n\'\\\'\')\n    sys.stderr.write("{} is not executable or does not exist in $PATH: {}".format(exe,PATH))\n    sys.exit(-1)\n\nif not True:\n    PR_SET_NO_NEW_PRIVS = 38\n    result = ctypes.CDLL(\'\\\'\'libc.so.6\'\\\'\').prctl(PR_SET_NO_NEW_PRIVS, 1, 0, 0, 0)\n\n    if result != 0:\n        sys.stdout.write(\'\\\'\'3\\n\'\\\'\')\n        sys.stdout.write("Could not disable setuid: prctl(PR_SET_NO_NEW_PRIVS) failed")\n        sys.exit(-1)\n\ntry:\n    PR_SET_PTRACER = 0x59616d61\n    PR_SET_PTRACER_ANY = -1\n    ctypes.CDLL(\'\\\'\'libc.so.6\'\\\'\').prctl(PR_SET_PTRACER, PR_SET_PTRACER_ANY, 0, 0, 0)\nexcept Exception:\n    pass\n\n# Determine what UID the process will execute as\n# This is used for locating apport core dumps\nsuid = os.getuid()\nsgid = os.getgid()\nst = os.stat(exe)\nif True:\n    if (st.st_mode & stat.S_ISUID):\n        suid = st.st_uid\n    if (st.st_mode & stat.S_ISGID):\n        sgid = st.st_gid\n\nif sys.argv[-1] == \'\\\'\'check\'\\\'\':\n    sys.stdout.write("1\\n")\n    sys.stdout.write(str(os.getpid()) + "\\n")\n    sys.stdout.write(str(os.getuid()) + "\\n")\n    sys.stdout.write(str(os.getgid()) + "\\n")\n    sys.stdout.write(str(suid) + "\\n")\n    sys.stdout.write(str(sgid) + "\\n")\n    sys.stdout.write(os.path.realpath(exe) + \'\\\'\'\\x00\'\\\'\')\n    sys.stdout.flush()\n\nfor fd, newfd in {0: 0, 1: 1, 2:2}.items():\n    if newfd is None:\n        close(fd)\n    elif isinstance(newfd, str):\n        os.close(fd)\n        os.open(newfd, os.O_RDONLY if fd == 0 else (os.O_RDWR|os.O_CREAT))\n    elif isinstance(newfd, int) and newfd != fd:\n        os.dup2(fd, newfd)\n\nif not True:\n    if platform.system().lower() == \'\\\'\'linux\'\\\'\' and True is not True:\n        ADDR_NO_RANDOMIZE = 0x0040000\n        ctypes.CDLL(\'\\\'\'libc.so.6\'\\\'\').personality(ADDR_NO_RANDOMIZE)\n\n    resource.setrlimit(resource.RLIMIT_STACK, (-1, -1))\n\n# Attempt to dump ALL core file regions\ntry:\n    with open(\'\\\'\'/proc/self/coredump_filter\'\\\'\', \'\\\'\'w\'\\\'\') as core_filter:\n        core_filter.write(\'\\\'\'0x3f\\n\'\\\'\')\nexcept Exception:\n    pass\n\n# Assume that the user would prefer to have core dumps.\ntry:\n    resource.setrlimit(resource.RLIMIT_CORE, (-1, -1))\nexcept Exception:\n    pass\n\ndef getppid():\n                import os\n                print(os.getppid())\napply(getppid, [])\n\nos.execve(exe, argv, os.environ)\' check; done; echo 2'
```

**Comment by zachriggle** [comment]
This was not happening as recent as https://travis-ci.org/Gallopsled/pwntools/builds/221838602

Which tested 348f6dfbe74d13787d80175ef45bb20a6926301b from April 13

**Comment by zachriggle** [comment]
It was also not happening with https://travis-ci.org/Gallopsled/pwntools/builds/231467318

Which tested b8e902b61197ab98628af19efadc07800dac2507 from May 12

**Comment by zachriggle** [comment]
From the list of builds... https://travis-ci.org/Gallopsled/pwntools/builds

f7568b47b3fd5c33d14c3dc600f6dd72a6af768e <-- fucked

2ccd36ca7f904f9973c3d19b2cf9c6e719a9b08e <-- fucked

9062360041927139922671d62458b6d40aa5e023 <-- okay

It looks like everything changed with 2ccd36ca7f904f9973c3d19b2cf9c6e719a9b08e, but that just changed the `CHANGELOG.md`.

I expect this is a Travis CI issue.

**Comment by zachriggle** [comment]
I re-started the build https://travis-ci.org/Gallopsled/pwntools/builds/232633297 for 9062360 to see what it does.

**Comment by zachriggle** [comment]
Yep it's Travis's fault.  They fucked logging somehow.

**Comment by zachriggle** [comment]
Pretty sure this was resolved with #1036
